test/e2e/daemonset_test.go

```diff
@@ -68,6 +68,7 @@ func (suite *DaemonSetTestSuite) TestDaemonSet()  {
 	var err error
 	cleanupOptions := &framework.CleanupOptions{TestContext: ctx, Timeout: timeout, RetryInterval: retryInterval}
 
+	j := jaegerAgentAsDaemonsetDefinition(namespace, "agent-as-daemonset")
```

Comment:
This isn't used in the openshift path - would it be better just to move back to else clause and have the `defer undeployJaegerInstance(j)` only occur in that path too?

Reply:
@objectiser I've submitted a fix for undeploy on OpenShift in the daemonset tests.  As far as waiting for an instance to be deleted, I can't say for sure if there is potential for overlapping instances.  @jpkrohling  what do you think?

I can add something to wait on undeploy, but it may end up being simpler just to go back to the other solution, and create a new namespace for each testcase.

Reply:
I think we do not have tests running in parallel anymore, since we started watching all namespaces, but I might be wrong. Running `kubectl get pods --all-namespaces` while the tests are running might provide the answer.

In any case, I'd say that most tests (especially `smoke`) should be prepared to be executed in parallel, while others (`es`) might assume that tests are executed serially.

Reply:
@objectiser @jpkrohling I've just submitted a change that will cause tests to wait until the jaeger instance is deleted.  I tried creating a namespace per test case, but immediately ran into problems because then we get longer namespace names (like allinonesuite-testallinonewithingress-1562847181 instead of allinonesuite-1562847181) and this causes other problems, like route names being to long.

It might be nice to do this in the future if it means we can parallelize tests, but I think it's probably out of scope for this change.